File: 242-valid-anagram/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s)!=len(t):
            return False
        hashmap={}
        for c in s:
            hashmap[c]=hashmap.get(c,0)+1
        for c in t:
            if c not in hashmap.keys():
                return False
            hashmap[c]-=1
            if hashmap[c]<0:
                return False

        return True

File: 242-valid-anagram/valid-anagram.cpp
#include <algorithm>
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size()!=t.size()) return false;
        unordered_map<char,int> mp;
        for (int i=0;i<s.size();i++){
            mp[s[i]]+=1;
        }
        for (int j=0;j<t.size();j++){
            if (mp.find(t[j])==mp.end()) return false;
            mp[t[j]]-=1;
            if (mp[t[j]]==0){
                mp.erase(t[j]);
            }
        }
        return mp.size()==0;
    }
};